#              VERZEO MINOR PROJECT
##                    Done by - Gowtham.V.R


# The given task at hand

* We have a dataset at hand that is "Tmdb-movies" that consists of multiple columns relating to movies and other important details regarding the said the movies over a certain period of time 

## Now, we have 5 questions, for which we need to find the answers for through EDA(Explaratory data analysis).

The questions are:
* Which are the movies with the third-lowest and third-highest budget?
* What is the average number of words in movie titles between the year 2000-2005?
* What is the most common Genre for Vin Diesel & Emma Watson movies?
* Which are the movies with most and least earned revenue?
* What is the average runtime of movies in the year 2006?

In [1]:
import pandas as pd
import numpy as np 

In [2]:
%matplotlib inline
import matplotlib.pyplot as plt

In [3]:
movie_df = pd.read_csv("tmdb-movies.csv")
movie_df

id    imdb_id  popularity     budget     revenue  \
0      135397  tt0369610   32.985763  150000000  1513528810   
1       76341  tt1392190   28.419936  150000000   378436354   
2      262500  tt2908446   13.112507  110000000   295238201   
3      140607  tt2488496   11.173104  200000000  2068178225   
4      168259  tt2820852    9.335014  190000000  1506249360   
...       ...        ...         ...        ...         ...   
10861      21  tt0060371    0.080598          0           0   
10862   20379  tt0060472    0.065543          0           0   
10863   39768  tt0060161    0.065141          0           0   
10864   21449  tt0061177    0.064317          0           0   
10865   22293  tt0060666    0.035919      19000           0   

                     original_title  \
0                    Jurassic World   
1                Mad Max: Fury Road   
2                         Insurgent   
3      Star Wars: The Force Awakens   
4                         Furious 7   
...                             ...   
10861            The Endless Summer   
10862                    Grand Prix   
10863           Beregis Avtomobilya   
10864        What's Up, Tiger Lily?   
10865      Manos: The Hands of Fate   

                                                    cast  \
0      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
2      Shailene Woodley|Theo James|Kate Winslet|Ansel...   
3      Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
4      Vin Diesel|Paul Walker|Jason Statham|Michelle ...   
...                                                  ...   
10861  Michael Hynson|Robert August|Lord 'Tally Ho' B...   
10862  James Garner|Eva Marie Saint|Yves Montand|Tosh...   
10863  Innokentiy Smoktunovskiy|Oleg Efremov|Georgi Z...   
10864  Tatsuya Mihashi|Akiko Wakabayashi|Mie Hama|Joh...   
10865  Harold P. Warren|Tom Neyman|John Reynolds|Dian...   

                                                homepage            director  \
0                          http://www.jurassicworld.com/     Colin Trevorrow   
1                            http://www.madmaxmovie.com/       George Miller   
2         http://www.thedivergentseries.movie/#insurgent    Robert Schwentke   
3      http://www.starwars.com/films/star-wars-episod...         J.J. Abrams   
4                               http://www.furious7.com/           James Wan   
...                                                  ...                 ...   
10861                                                NaN         Bruce Brown   
10862                                                NaN  John Frankenheimer   
10863                                                NaN      Eldar Ryazanov   
10864                                                NaN         Woody Allen   
10865                                                NaN    Harold P. Warren   

                                                 tagline  ...  \
0                                      The park is open.  ...   
1                                     What a Lovely Day.  ...   
2                             One Choice Can Destroy You  ...   
3                          Every generation has a story.  ...   
4                                    Vengeance Hits Home  ...   
...                                                  ...  ...   
10861                                                NaN  ...   
10862  Cinerama sweeps YOU into a drama of speed and ...  ...   
10863                                                NaN  ...   
10864                          WOODY ALLEN STRIKES BACK!  ...   
10865       It's Shocking! It's Beyond Your Imagination!  ...   

                                                overview runtime  \
0      Twenty-two years after the events of Jurassic ...     124   
1      An apocalyptic story set in the furthest reach...     120   
2      Beatrice Prior must confront her inner demons ...     119   
3      Thirty years after defeating the Galactic E

In [4]:
movie_df.dtypes

id                        int64
imdb_id                  object
popularity              float64
budget                    int64
revenue                   int64
original_title           object
cast                     object
homepage                 object
director                 object
tagline                  object
keywords                 object
overview                 object
runtime                   int64
genres                   object
production_companies     object
release_date             object
vote_count                int64
vote_average            float64
release_year              int64
budget_adj              float64
revenue_adj             float64
dtype: object

In [5]:
movie_df.describe()

id    popularity        budget       revenue       runtime  \
count   10866.000000  10866.000000  1.086600e+04  1.086600e+04  10866.000000   
mean    66064.177434      0.646441  1.462570e+07  3.982332e+07    102.070863   
std     92130.136561      1.000185  3.091321e+07  1.170035e+08     31.381405   
min         5.000000      0.000065  0.000000e+00  0.000000e+00      0.000000   
25%     10596.250000      0.207583  0.000000e+00  0.000000e+00     90.000000   
50%     20669.000000      0.383856  0.000000e+00  0.000000e+00     99.000000   
75%     75610.000000      0.713817  1.500000e+07  2.400000e+07    111.000000   
max    417859.000000     32.985763  4.250000e+08  2.781506e+09    900.000000   

         vote_count  vote_average  release_year    budget_adj   revenue_adj  
count  10866.000000  10866.000000  10866.000000  1.086600e+04  1.086600e+04  
mean     217.389748      5.974922   2001.322658  1.755104e+07  5.136436e+07  
std      575.619058      0.935142     12.812941  3.430616e+07  1.446325e+08  
min       10.000000      1.500000   1960.000000  0.000000e+00  0.000000e+00  
25%       17.000000      5.400000   1995.000000  0.000000e+00  0.000000e+00  
50%       38.000000      6.000000   2006.000000  0.000000e+00  0.000000e+00  
75%      145.750000      6.600000   2011.000000  2.085325e+07  3.369710e+07  
max     9767.000000      9.200000   2015.000000  4.250000e+08  2.827124e+09

In [6]:
movie_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10866 entries, 0 to 10865
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   id                    10866 non-null  int64  
 1   imdb_id               10856 non-null  object 
 2   popularity            10866 non-null  float64
 3   budget                10866 non-null  int64  
 4   revenue               10866 non-null  int64  
 5   original_title        10866 non-null  object 
 6   cast                  10790 non-null  object 
 7   homepage              2936 non-null   object 
 8   director              10822 non-null  object 
 9   tagline               8042 non-null   object 
 10  keywords              9373 non-null   object 
 11  overview              10862 non-null  object 
 12  runtime               10866 non-null  int64  
 13  genres                10843 non-null  object 
 14  production_companies  9836 non-null   object 
 15  release_date       

In [7]:
movie_df.mean()

id              6.606418e+04
popularity      6.464410e-01
budget          1.462570e+07
revenue         3.982332e+07
runtime         1.020709e+02
vote_count      2.173897e+02
vote_average    5.974922e+00
release_year    2.001323e+03
budget_adj      1.755104e+07
revenue_adj     5.136436e+07
dtype: float64

In [8]:
movie_df.columns

Index(['id', 'imdb_id', 'popularity', 'budget', 'revenue', 'original_title',
       'cast', 'homepage', 'director', 'tagline', 'keywords', 'overview',
       'runtime', 'genres', 'production_companies', 'release_date',
       'vote_count', 'vote_average', 'release_year', 'budget_adj',
       'revenue_adj'],
      dtype='object')

### Now that we have laid out all the data from the csv file, we can now get to work on the questions. 

# 1. Which are the movies with the third-lowest and third-highest budget?

The first case is with column "budget"

In [9]:
df1 = movie_df[['budget','original_title']]
df1

budget                original_title
0      150000000                Jurassic World
1      150000000            Mad Max: Fury Road
2      110000000                     Insurgent
3      200000000  Star Wars: The Force Awakens
4      190000000                     Furious 7
...          ...                           ...
10861          0            The Endless Summer
10862          0                    Grand Prix
10863          0           Beregis Avtomobilya
10864          0        What's Up, Tiger Lily?
10865      19000      Manos: The Hands of Fate

[10866 rows x 2 columns]

In [10]:
df_budget = df1[df1['budget'] > 0]
df_budget

budget                original_title
0      150000000                Jurassic World
1      150000000            Mad Max: Fury Road
2      110000000                     Insurgent
3      200000000  Star Wars: The Force Awakens
4      190000000                     Furious 7
...          ...                           ...
10835   12000000              The Sand Pebbles
10841      75000                  The Shooting
10848    5115000              Fantastic Voyage
10855     700000       The Ghost & Mr. Chicken
10865      19000      Manos: The Hands of Fate

[5170 rows x 2 columns]

In [11]:
uv = df_budget["budget"].nunique()
print(uv)

556


In [12]:
df_budget.sort_values(by=['budget'], inplace=True)
df_budget

<ipython-input-12-3ee16baa8958>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_budget.sort_values(by=['budget'], inplace=True)


budget                               original_title
1151          1                                  Fear Clinic
3581          1                      Love, Wedding, Marriage
3065          1           Angus, Thongs and Perfect Snogging
2618          1                                 Lost & Found
8944          2                                 Death Wish 2
...         ...                                          ...
6570  270000000                             Superman Returns
14    280000000                      Avengers: Age of Ultron
7387  300000000     Pirates of the Caribbean: At World's End
3375  380000000  Pirates of the Caribbean: On Stranger Tides
2244  425000000                            The Warrior's Way

[5170 rows x 2 columns]

In [13]:
df_budget.head(10)              

budget                      original_title
1151        1                         Fear Clinic
3581        1             Love, Wedding, Marriage
3065        1  Angus, Thongs and Perfect Snogging
2618        1                        Lost & Found
8944        2                        Death Wish 2
3765        3                Death of a Superhero
10050       3  Tales from the Darkside: The Movie
2398        3                                 Boy
1313        5         Des Lendemains Qui Chantent
8235        6                 Tales from the Hood

### Now considering the fact we removed all the movies with budget as 0, the movie with the 3rd lowest budget is "DEATH WISH 2".

### If we don't consider the removed 0's, then there are three movies and they are

* Death of a Superhero
* Tales from the Darkside: The movie
* Boy

In [14]:
df_budget.tail(10)

budget                               original_title
643   250000000                   X-Men: Days of Future Past
5508  255000000                              The Lone Ranger
7394  258000000                                 Spider-Man 3
4411  260000000                                  John Carter
1929  260000000                                      Tangled
6570  270000000                             Superman Returns
14    280000000                      Avengers: Age of Ultron
7387  300000000     Pirates of the Caribbean: At World's End
3375  380000000  Pirates of the Caribbean: On Stranger Tides
2244  425000000                            The Warrior's Way

### The movie with the third highest budget is "Pirates of the Caribbean: At World's End"

Now the second case is with the column "budget_adj"

In [15]:
df1 = movie_df[['budget_adj','original_title']]
df1

budget_adj                original_title
0      1.379999e+08                Jurassic World
1      1.379999e+08            Mad Max: Fury Road
2      1.012000e+08                     Insurgent
3      1.839999e+08  Star Wars: The Force Awakens
4      1.747999e+08                     Furious 7
...             ...                           ...
10861  0.000000e+00            The Endless Summer
10862  0.000000e+00                    Grand Prix
10863  0.000000e+00           Beregis Avtomobilya
10864  0.000000e+00        What's Up, Tiger Lily?
10865  1.276423e+05      Manos: The Hands of Fate

[10866 rows x 2 columns]

In [16]:
df_budget = df1[df1['budget_adj'] > 0]
df_budget

budget_adj                original_title
0      1.379999e+08                Jurassic World
1      1.379999e+08            Mad Max: Fury Road
2      1.012000e+08                     Insurgent
3      1.839999e+08  Star Wars: The Force Awakens
4      1.747999e+08                     Furious 7
...             ...                           ...
10835  8.061618e+07              The Sand Pebbles
10841  5.038511e+05                  The Shooting
10848  3.436265e+07              Fantastic Voyage
10855  4.702610e+06       The Ghost & Mr. Chicken
10865  1.276423e+05      Manos: The Hands of Fate

[5170 rows x 2 columns]

In [17]:
df_budget.sort_values(by=['budget_adj'], inplace=True)
df_budget

<ipython-input-17-0d2ffbbea8df>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_budget.sort_values(by=['budget_adj'], inplace=True)


budget_adj                               original_title
1151  9.210911e-01                                  Fear Clinic
3581  9.693980e-01                      Love, Wedding, Marriage
3065  1.012787e+00           Angus, Thongs and Perfect Snogging
2618  1.309053e+00                                 Lost & Found
3765  2.908194e+00                         Death of a Superhero
...            ...                                          ...
5231  2.716921e+08                                      Titanic
6570  2.920507e+08                             Superman Returns
7387  3.155006e+08     Pirates of the Caribbean: At World's End
3375  3.683713e+08  Pirates of the Caribbean: On Stranger Tides
2244  4.250000e+08                            The Warrior's Way

[5170 rows x 2 columns]

In [18]:
df_budget.head(10)

budget_adj                      original_title
1151     0.921091                         Fear Clinic
3581     0.969398             Love, Wedding, Marriage
3065     1.012787  Angus, Thongs and Perfect Snogging
2618     1.309053                        Lost & Found
3765     2.908194                Death of a Superhero
2398     3.000000                                 Boy
8944     4.519285                        Death Wish 2
1313     4.605455         Des Lendemains Qui Chantent
10050    5.006696  Tales from the Darkside: The Movie
3015     8.102293                           Max Manus

In [19]:
df_budget.tail()

budget_adj                               original_title
5231  2.716921e+08                                      Titanic
6570  2.920507e+08                             Superman Returns
7387  3.155006e+08     Pirates of the Caribbean: At World's End
3375  3.683713e+08  Pirates of the Caribbean: On Stranger Tides
2244  4.250000e+08                            The Warrior's Way

# 2.What is the average number of words in movie titles between the year 2000-2005?

In [20]:
df2 = movie_df[['original_title','release_year']]
df2

original_title  release_year
0                    Jurassic World          2015
1                Mad Max: Fury Road          2015
2                         Insurgent          2015
3      Star Wars: The Force Awakens          2015
4                         Furious 7          2015
...                             ...           ...
10861            The Endless Summer          1966
10862                    Grand Prix          1966
10863           Beregis Avtomobilya          1966
10864        What's Up, Tiger Lily?          1966
10865      Manos: The Hands of Fate          1966

[10866 rows x 2 columns]

In [21]:
df2.sort_values(by=['release_year'], inplace=True)
df2

<ipython-input-21-c5890b7ac171>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2.sort_values(by=['release_year'], inplace=True)


original_title  release_year
10172                 Exodus          1960
10141                 Psycho          1960
10142  The Magnificent Seven          1960
10143              Spartacus          1960
10144          The Apartment          1960
...                      ...           ...
417           Pound of Flesh          2015
416                    Curve          2015
415           Show Me a Hero          2015
422          Do You Believe?          2015
0             Jurassic World          2015

[10866 rows x 2 columns]

In [22]:
df2 = df2[df2['release_year'] >= 2000]
df2

original_title  release_year
8718              The Cell          2000
8717       Mission to Mars          2000
8711  The Whole Nine Yards          2000
8716            Red Planet          2000
8715         High Fidelity          2000
...                    ...           ...
417         Pound of Flesh          2015
416                  Curve          2015
415         Show Me a Hero          2015
422        Do You Believe?          2015
0           Jurassic World          2015

[7168 rows x 2 columns]

In [23]:
df2 = df2[df2['release_year'] <= 2005]
df2

original_title  release_year
8718                 The Cell          2000
8717          Mission to Mars          2000
8711     The Whole Nine Yards          2000
8716               Red Planet          2000
8715            High Fidelity          2000
...                       ...           ...
6346              White Noise          2005
6345  Goal!: The Dream Begins          2005
6352                Manderlay          2005
6542        Un amour à taire          2005
6401               Earthlings          2005

[1687 rows x 2 columns]

In [24]:
df2['count'] = df2['original_title'].str.split().str.len()
df2

<ipython-input-24-48a3510b67e2>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['count'] = df2['original_title'].str.split().str.len()


original_title  release_year  count
8718                 The Cell          2000      2
8717          Mission to Mars          2000      3
8711     The Whole Nine Yards          2000      4
8716               Red Planet          2000      2
8715            High Fidelity          2000      2
...                       ...           ...    ...
6346              White Noise          2005      2
6345  Goal!: The Dream Begins          2005      4
6352                Manderlay          2005      1
6542        Un amour à taire          2005      4
6401               Earthlings          2005      1

[1687 rows x 3 columns]

In [25]:
Sum = df2['count'].sum()
print(Sum)

4858


In [26]:
average_words = 4858/1687
average_words

2.879668049792531

### So, the average number of words is 2.879668049792531, which is approximately around 3 words per movie title.

# 3) What is the most common Genre for Vin Diesel & Emma Watson movies?

In [27]:
df3 = movie_df[['cast','genres']]
df3

cast  \
0      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
2      Shailene Woodley|Theo James|Kate Winslet|Ansel...   
3      Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
4      Vin Diesel|Paul Walker|Jason Statham|Michelle ...   
...                                                  ...   
10861  Michael Hynson|Robert August|Lord 'Tally Ho' B...   
10862  James Garner|Eva Marie Saint|Yves Montand|Tosh...   
10863  Innokentiy Smoktunovskiy|Oleg Efremov|Georgi Z...   
10864  Tatsuya Mihashi|Akiko Wakabayashi|Mie Hama|Joh...   
10865  Harold P. Warren|Tom Neyman|John Reynolds|Dian...   

                                          genres  
0      Action|Adventure|Science Fiction|Thriller  
1      Action|Adventure|Science Fiction|Thriller  
2             Adventure|Science Fiction|Thriller  
3       Action|Adventure|Science Fiction|Fantasy  
4                          Action|Crime|Thriller  
...                                          ...  
10861                                Documentary  
10862                     Action|Adventure|Drama  
10863                             Mystery|Comedy  
10864                              Action|Comedy  
10865                                     Horror  

[10866 rows x 2 columns]

In [28]:
sub = 'Vin Diesel'
df3["Indexes"] = df3["cast"].str.find(sub)
df3

<ipython-input-28-c4e24908df99>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df3["Indexes"] = df3["cast"].str.find(sub)


cast  \
0      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
2      Shailene Woodley|Theo James|Kate Winslet|Ansel...   
3      Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
4      Vin Diesel|Paul Walker|Jason Statham|Michelle ...   
...                                                  ...   
10861  Michael Hynson|Robert August|Lord 'Tally Ho' B...   
10862  James Garner|Eva Marie Saint|Yves Montand|Tosh...   
10863  Innokentiy Smoktunovskiy|Oleg Efremov|Georgi Z...   
10864  Tatsuya Mihashi|Akiko Wakabayashi|Mie Hama|Joh...   
10865  Harold P. Warren|Tom Neyman|John Reynolds|Dian...   

                                          genres  Indexes  
0      Action|Adventure|Science Fiction|Thriller     -1.0  
1      Action|Adventure|Science Fiction|Thriller     -1.0  
2             Adventure|Science Fiction|Thriller     -1.0  
3       Action|Adventure|Science Fiction|Fantasy     -1.0  
4                          Action|Crime|Thriller      0.0  
...                                          ...      ...  
10861                                Documentary     -1.0  
10862                     Action|Adventure|Drama     -1.0  
10863                             Mystery|Comedy     -1.0  
10864                              Action|Comedy     -1.0  
10865                                     Horror     -1.0  

[10866 rows x 3 columns]

In [29]:
df3vd = df3[df3['Indexes'] == 0]
df3vd

cast  \
4     Vin Diesel|Paul Walker|Jason Statham|Michelle ...   
41    Vin Diesel|Rose Leslie|Michael Caine|Elijah Wo...   
1539  Vin Diesel|Paul Walker|Jordana Brewster|Michel...   
1907  Vin Diesel|Michelle Rodriguez|Sung Kang|Don Om...   
2942  Vin Diesel|Michelle Yeoh|Mélanie Thierry|Lamb...   
3619  Vin Diesel|Paul Walker|Dwayne Johnson|Jordana ...   
3924  Vin Diesel|Asia Argento|Marton Csokas|Samuel L...   
4974  Vin Diesel|Larenz Tate|Timothy Olyphant|Geno S...   
5437  Vin Diesel|Karl Urban|Katee Sackhoff|Jordi Mol...   
5793                              Vin Diesel|Karl Urban   
6204  Vin Diesel|Lauren Graham|Faith Ford|Chris Pott...   
6774  Vin Diesel|Alex Rocco|Frank Pietrangolare|Rich...   
7024  Vin Diesel|Christina Cox|Colm Feore|Karl Urban...   
7208  Vin Diesel|Rhiana Griffith|Keith David|Nick Ch...   
8684  Vin Diesel|Radha Mitchell|Cole Hauser|Rhiana G...   

                                         genres  Indexes  
4                         Action|Crime|Thriller      0.0  
41                     Fantasy|Action|Adventure      0.0  
1539                Action|Crime|Drama|Thriller      0.0  
1907                               Action|Drama      0.0  
2942  Action|Adventure|Science Fiction|Thriller      0.0  
3619                      Action|Thriller|Crime      0.0  
3924                                     Action      0.0  
4974                               Action|Drama      0.0  
5437            Science Fiction|Action|Thriller      0.0  
5793           Action|Animation|Science Fiction      0.0  
6204        Action|Comedy|Drama|Family|Thriller      0.0  
6774                  Drama|Action|Comedy|Crime      0.0  
7024  Adventure|Action|Thriller|Science Fiction      0.0  
7208  Action|Animation|Science Fiction|Thriller      0.0  
8684            Thriller|Science Fiction|Action      0.0

In [30]:
df3vd.count()

cast       15
genres     15
Indexes    15
dtype: int64

In [31]:
g_and_c = {}
for i in range(df3vd.shape[0]):
    genres = str(df3vd['genres'].values[i]).split('|')
    for j in genres:
        try:
            count = g_and_c[j]
            g_and_c[j] = count + 1
        except:
            g_and_c[j] = 1
        
g_and_c

{'Action': 15,
 'Crime': 4,
 'Thriller': 9,
 'Fantasy': 1,
 'Adventure': 3,
 'Drama': 5,
 'Science Fiction': 6,
 'Animation': 2,
 'Comedy': 2,
 'Family': 1}

### From this we can conclude that the most common genre for Vin Diesel is action with 15 entries

### Now for Emma Watson,

In [32]:
df3 = movie_df[['cast','genres']]
df3

cast  \
0      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
2      Shailene Woodley|Theo James|Kate Winslet|Ansel...   
3      Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
4      Vin Diesel|Paul Walker|Jason Statham|Michelle ...   
...                                                  ...   
10861  Michael Hynson|Robert August|Lord 'Tally Ho' B...   
10862  James Garner|Eva Marie Saint|Yves Montand|Tosh...   
10863  Innokentiy Smoktunovskiy|Oleg Efremov|Georgi Z...   
10864  Tatsuya Mihashi|Akiko Wakabayashi|Mie Hama|Joh...   
10865  Harold P. Warren|Tom Neyman|John Reynolds|Dian...   

                                          genres  
0      Action|Adventure|Science Fiction|Thriller  
1      Action|Adventure|Science Fiction|Thriller  
2             Adventure|Science Fiction|Thriller  
3       Action|Adventure|Science Fiction|Fantasy  
4                          Action|Crime|Thriller  
...                                          ...  
10861                                Documentary  
10862                     Action|Adventure|Drama  
10863                             Mystery|Comedy  
10864                              Action|Comedy  
10865                                     Horror  

[10866 rows x 2 columns]

In [33]:
sub = 'Emma Watson'
df3["Indexes"] = df3["cast"].str.find(sub)
df3

<ipython-input-33-520c43a853ba>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df3["Indexes"] = df3["cast"].str.find(sub)


cast  \
0      Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1      Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
2      Shailene Woodley|Theo James|Kate Winslet|Ansel...   
3      Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
4      Vin Diesel|Paul Walker|Jason Statham|Michelle ...   
...                                                  ...   
10861  Michael Hynson|Robert August|Lord 'Tally Ho' B...   
10862  James Garner|Eva Marie Saint|Yves Montand|Tosh...   
10863  Innokentiy Smoktunovskiy|Oleg Efremov|Georgi Z...   
10864  Tatsuya Mihashi|Akiko Wakabayashi|Mie Hama|Joh...   
10865  Harold P. Warren|Tom Neyman|John Reynolds|Dian...   

                                          genres  Indexes  
0      Action|Adventure|Science Fiction|Thriller     -1.0  
1      Action|Adventure|Science Fiction|Thriller     -1.0  
2             Adventure|Science Fiction|Thriller     -1.0  
3       Action|Adventure|Science Fiction|Fantasy     -1.0  
4                          Action|Crime|Thriller     -1.0  
...                                          ...      ...  
10861                                Documentary     -1.0  
10862                     Action|Adventure|Drama     -1.0  
10863                             Mystery|Comedy     -1.0  
10864                              Action|Comedy     -1.0  
10865                                     Horror     -1.0  

[10866 rows x 3 columns]

In [34]:
df3ew = df3[df3['Indexes'] == 0]
df3ew

cast        genres  Indexes
5546  Emma Watson|Israel Broussard|Taissa Farmiga|Ka...   Drama|Crime      0.0
7786  Emma Watson|Yasmin Paige|Lucy Boynton|Emilia F...  Drama|Family      0.0

In [35]:
df3ew.count()

cast       2
genres     2
Indexes    2
dtype: int64

In [36]:
g_and_c = {}
for i in range(df3ew.shape[0]):
    genres = str(df3ew['genres'].values[i]).split('|')
    for j in genres:
        try:
            count = g_and_c[j]
            g_and_c[j] = count + 1
        except:
            g_and_c[j] = 1
        
g_and_c

{'Drama': 2, 'Crime': 1, 'Family': 1}

### From this we can conclude that the most common genre for Emma Watson is Drama with 2 entries

# 4) Which are the movies with most and least earned revenue?

The first case is with revenue column

In [37]:
df4 = movie_df[['revenue','original_title']]
df4

revenue                original_title
0      1513528810                Jurassic World
1       378436354            Mad Max: Fury Road
2       295238201                     Insurgent
3      2068178225  Star Wars: The Force Awakens
4      1506249360                     Furious 7
...           ...                           ...
10861           0            The Endless Summer
10862           0                    Grand Prix
10863           0           Beregis Avtomobilya
10864           0        What's Up, Tiger Lily?
10865           0      Manos: The Hands of Fate

[10866 rows x 2 columns]

In [38]:
df4.sort_values(by=['revenue'], inplace=True)
df4

<ipython-input-38-bcc5bcdd39f0>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df4.sort_values(by=['revenue'], inplace=True)


revenue                                     original_title
10865           0                           Manos: The Hands of Fate
4553            0  A Turtle’s Tale 2: Sammy’s Escape From Par...
4552            0                                      Truth or Dare
4551            0                                   Laurence Anyways
4549            0                                            Io e te
...           ...                                                ...
0      1513528810                                     Jurassic World
4361   1519557910                                       The Avengers
5231   1845034188                                            Titanic
3      2068178225                       Star Wars: The Force Awakens
1386   2781505847                                             Avatar

[10866 rows x 2 columns]

In [39]:
df4.head(6017)        

revenue                                     original_title
10865        0                           Manos: The Hands of Fate
4553         0  A Turtle’s Tale 2: Sammy’s Escape From Par...
4552         0                                      Truth or Dare
4551         0                                   Laurence Anyways
4549         0                                            Io e te
...        ...                                                ...
2271         0                                The Four-Faced Liar
1891         0                  Nightmares in Red, White and Blue
10049        0                                  Lord of the Flies
1972         0                         Undisputed III: Redemption
8142         2                                           Mallrats

[6017 rows x 2 columns]

In [40]:
df4.tail(5)

revenue                original_title
0     1513528810                Jurassic World
4361  1519557910                  The Avengers
5231  1845034188                       Titanic
3     2068178225  Star Wars: The Force Awakens
1386  2781505847                        Avatar

### So, there are precisely 6016 movie titles with their revenue as 0

### The next movie with least revenue earned is Mallrats with 2 dollars earned

### The movie with the most earned revenue is Avatar with 2781505847 dollars earned in the box office	

The second scenario is with the column "revenue_adj"

In [41]:
df4 = movie_df[['revenue_adj','original_title']]
df4

revenue_adj                original_title
0      1.392446e+09                Jurassic World
1      3.481613e+08            Mad Max: Fury Road
2      2.716190e+08                     Insurgent
3      1.902723e+09  Star Wars: The Force Awakens
4      1.385749e+09                     Furious 7
...             ...                           ...
10861  0.000000e+00            The Endless Summer
10862  0.000000e+00                    Grand Prix
10863  0.000000e+00           Beregis Avtomobilya
10864  0.000000e+00        What's Up, Tiger Lily?
10865  0.000000e+00      Manos: The Hands of Fate

[10866 rows x 2 columns]

In [42]:
df4.sort_values(by=['revenue_adj'], inplace=True)
df4

<ipython-input-42-3a3574523b02>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df4.sort_values(by=['revenue_adj'], inplace=True)


revenue_adj                             original_title
10865  0.000000e+00                   Manos: The Hands of Fate
5132   0.000000e+00                                   Kontroll
5129   0.000000e+00                                   Dead End
5127   0.000000e+00                              The Statement
5126   0.000000e+00  Scooby-Doo! And the Legend of the Vampire
...             ...                                        ...
9806   1.907006e+09                                       Jaws
10594  2.167325e+09                               The Exorcist
5231   2.506406e+09                                    Titanic
1329   2.789712e+09                                  Star Wars
1386   2.827124e+09                                     Avatar

[10866 rows x 2 columns]

In [43]:
df4.tail(10)

revenue_adj                  original_title
4361   1.443191e+09                    The Avengers
10110  1.574815e+09  One Hundred and One Dalmatians
8094   1.583050e+09                         The Net
8889   1.791694e+09      E.T. the Extra-Terrestrial
3      1.902723e+09    Star Wars: The Force Awakens
9806   1.907006e+09                            Jaws
10594  2.167325e+09                    The Exorcist
5231   2.506406e+09                         Titanic
1329   2.789712e+09                       Star Wars
1386   2.827124e+09                          Avatar

In [44]:
df4i = df4[df4['revenue_adj'] > 0]
df4i

revenue_adj                  original_title
5067   2.370705e+00                 Shattered Glass
8142   2.861934e+00                        Mallrats
3239   3.038360e+00  Dr. Horrible's Sing-Along Blog
5162   5.926763e+00                     Kid's Story
8523   6.951084e+00               Bordello of Blood
...             ...                             ...
9806   1.907006e+09                            Jaws
10594  2.167325e+09                    The Exorcist
5231   2.506406e+09                         Titanic
1329   2.789712e+09                       Star Wars
1386   2.827124e+09                          Avatar

[4850 rows x 2 columns]

In [45]:
df4i.sort_values(by=['revenue_adj'], inplace=True)
df4i

<ipython-input-45-25ee6fc56191>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df4i.sort_values(by=['revenue_adj'], inplace=True)


revenue_adj                  original_title
5067   2.370705e+00                 Shattered Glass
8142   2.861934e+00                        Mallrats
3239   3.038360e+00  Dr. Horrible's Sing-Along Blog
5162   5.926763e+00                     Kid's Story
8523   6.951084e+00               Bordello of Blood
...             ...                             ...
9806   1.907006e+09                            Jaws
10594  2.167325e+09                    The Exorcist
5231   2.506406e+09                         Titanic
1329   2.789712e+09                       Star Wars
1386   2.827124e+09                          Avatar

[4850 rows x 2 columns]

### The movie with the highest revenue is avatar

### The movie with the least revenue when all 0's are removed is Shattered Glass

# 5) What is the average runtime of movies in the year 2006?

In [46]:
df5 = movie_df[['runtime','release_year']]
df5

runtime  release_year
0          124          2015
1          120          2015
2          119          2015
3          136          2015
4          137          2015
...        ...           ...
10861       95          1966
10862      176          1966
10863       94          1966
10864       80          1966
10865       74          1966

[10866 rows x 2 columns]

In [47]:
df5 = df5[df5['release_year'] == 2006]
df5

runtime  release_year
6554      106          2006
6555      151          2006
6556      117          2006
6557      144          2006
6558      149          2006
...       ...           ...
6957      107          2006
6958      100          2006
6959       98          2006
6960       92          2006
6961      135          2006

[408 rows x 2 columns]

In [48]:
Runtime = df5['runtime'].sum()
print(Runtime)

41487


In [49]:
avg_runtime = 41487/408
print(avg_runtime)

101.68382352941177


### The average runtime of movies in the year 2006 is 101.68382352941177 minutes